lib/src/main/java/com/auth0/jwt/JWTVerifier.java

```diff
@@ -127,49 +133,54 @@ public Verification ignoreIssuedAt() {
 
         @Override
         public Verification withJWTId(String jwtId) {
-            requireClaim(PublicClaims.JWT_ID, jwtId);
+            checkIfNeedToRemove(PublicClaims.JWT_ID, jwtId, ((claim, decodedJWT) -> jwtId.equals(claim.asString())));
             return this;
         }
 
         @Override
         public Verification withClaimPresence(String name) throws IllegalArgumentException {
             assertNonNull(name);
-            requireClaim(name, NonEmptyClaim.getInstance());
+            withClaim(name, ((claim, decodedJWT) -> {
+                if (claim instanceof NullClaim) {
```

Comment:
Can't we just use `claim.isNull()` ?

Reply:
`withClaimPresence` method is used to check whether a claim in the given name is present.  Our next task will be handling `null` values in our JWT (SDK-3158). Since in that task we will be acknowledging the Claim's presence in JWT (i.e not throw Exception as in here) this method's behaviour will change in that task. I guess it would be better for us to handle it there.